pdf-email/functions/pdf.js

```diff
@@ -0,0 +1,127 @@
+                                ]
+                            };
+
+      var pdfDoc = new pdfPrinter({
+      Roboto: {normal: new Buffer(require('pdfmake/build/vfs_fonts.js').pdfMake.vfs['Roboto-Regular.ttf'], 'base64')}
+      }).createPdfKitDocument(docDefinition);
+      let temp = "/tmp/";
+      let fileName = 'receipt.pdf';
+      let path = temp + fileName;
+      pdfDoc.pipe(fs.createWriteStream(path));
+      pdfDoc.end();
+
+
+  /*Find the file in the /tmp/ folder and provide the appropriate permissions*/
+    var cmd=require('node-cmd');
+    cmd.get('chmod +x /tmp/' + fileName, function(err, data, stderr){
+        console.log("Error: " + stderr);
+        console.log("Added +x flag to the file");
+
+        /*Show the contents. This is just for you to see the file, not really affecting the end result, you can remove it*/
+        cmd.get('ls -l /tmp/', function(err, data, stderr){
+        console.log("Error: " + stderr);
+        console.log("Data: " + data);
+
+        /*Prepare the file. Twilio Sendgrid requires the file to be on base64 format*/
+        const request = require('request');
+        const pdf2base64 = require('pdf-to-base64');
+        pdf2base64(path)
+        .then(
+            (response) => {
+
+            let data =
+              {
+              "personalizations": [
+                {
+                  "to": [
+                    {
+                      "email": context.TO_EMAIL_ADDRESS,
+                      "name": "TO Name"
+                    }
+                  ],
+                  "subject": "Hello! Here is your pdf"
+                }
+              ],
+              "from": {
+                "email": context.FROM_EMAIL_ADDRESS
+              },
+              "content": [
+                {
+                  "type": "text/plain",
+                  "value": "Here is the content of your email"
+                }
+              ],
+              "attachments": [
+                {
+                  "filename": fileName,
+                  "content": response
+                }
+              ]//,
+              //OPTIONAL IF YOU HAVE YOUR OWN template_id
+              //"template_id": context.EMAIL_TEMPLATE_ID
+          };
+
+
+        /*Send the request*/
+            request({
+                  url: "https://api.sendgrid.com/v3/mail/send",
+                  method: "POST",
+                  headers: {
+                    "Content-Type": "application/json"
+                  },
+                  auth: {
+                      "bearer": context.SENDGRID_API_KEY
+                  },
+                  json: data
+                }, function (error, response, body){
+
+                      console.log(error);
+                      console.log(JSON.stringify(response));
+                      console.log(body);
+                     callback();
+                });
+
+
+          })
+          .catch(
+              (error) => {
+                  console.log(error);
```

Comment:
Here you want to swap it with `callback(error)` which will make the Function fail and add it to the Twilio Debugger which is what you would want in this situation

Reply:
done